RoboticonColony/Assets/Code/Roboticon.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// A Roboticon. Holds all information about the roboticon and facilitates upgrading said roboticon. 
+/// </summary>
+sealed public class Roboticon {
+
+    /// <summary>
+    /// The current Tile that the roboticon is assigned to. 
+    /// Also allows reference to itself to read, not write. 
+    /// </summary>
+    public Tile CurrentTile { get; private set; }
+
+    /// <summary>
+    /// _bonusProduction is a dictionary which holds the int value (Total production of each recource) it produces.
+    /// </summary>
+    private Dictionary<ItemType, int> _bonusProduction;
+
+    /// <summary>
+    /// CurrentCustomisations is a list of RoboticonCustomisations which the roboticon currently has applied. 
+    /// Also allows reference to itself to read, not write. 

```

Comment:
If by this you mean you have get but private set you probably shouldn't include this in the documentation. as the set is private it simply wont appear when someone is using the class and an error will be thrown if someone does try to set. its also quite obvious that you can read but not write from the code.